ResearchKit/Common/ORKImageCaptureView.m

```diff
@@ -156,35 +149,57 @@ - (void)setError:(NSError * __nullable)error {
 const CGFloat CONTINUE_ALPHA_OPAQUE = 0;
 
 - (void)updateConstraints {
+    if (_variableConstraints) {
```

Comment:
From `updateConstraints` documentation:

    You must not invalidate any constraints as part of your constraint update phase. You also must not invoke a layout or drawing phase as part of constraint updating.

I think that *invalidating constraints* means doing anything that would potentially call `setNeedsUpdateConstraints`. I'm assuming that *invalidate* != *deactivate*, so I figured we can remove or deactivate constraints in `updateConstraints`. In addition, *ResearchKit* is already removing constraints this way in several other places and it doesn't seem to be a problem.

---

But I agree that the docs aren't very clear on this. Could you possibly get an authoritative answer?